test/luatest_helpers/cluster.lua

```diff
@@ -72,18 +72,37 @@ function Cluster:stop()
 end
 
 function Cluster:start(opts)
-    for _, server in ipairs(self.servers) do
-        if not server.process then
-            server:start({wait_for_readiness = false})
-        end
-    end
-
     -- The option is true by default.
     local wait_for_readiness = true
     if opts ~= nil and opts.wait_for_readiness ~= nil then
         wait_for_readiness = opts.wait_for_readiness
     end
 
+    -- The option is false by default.
+    local sequential_startup = false
+    if opts ~= nil and opts.sequential_startup ~= nil then
+        sequential_startup = opts.sequential_startup
+    end
+
+    local replication_settings = {}
+
+    for i, server in ipairs(self.servers) do
+        if not server.process then
+            if sequential_startup then
+                replication_settings[i] = server.box_cfg.replication
+                server.box_cfg.replication = self.servers[1].net_box_uri
+            end
+            server:start({wait_for_readiness = sequential_startup})
+        end
+    end
+
+    if sequential_startup then
+        for i, server in ipairs(self.servers) do
+            server:box_config({replication = replication_settings[i]})
```

Comment:
I guess you misunderstood my previous comment.
Imagine you have a cluster, where one server is running, and others are stopped. Then someone does `cluster:start()` on the cluster. This will nullify the already-running server's `box.cfg.replication`, because `replicatoin_settings[i]` will be `nil` for it.

So the proposed change is
```diff
diff --git a/test/luatest_helpers/cluster.lua b/test/luatest_helpers/cluster.lua
index 0d0390691..ed0ae8b20 100644
--- a/test/luatest_helpers/cluster.lua
+++ b/test/luatest_helpers/cluster.lua
@@ -98,8 +98,10 @@ function Cluster:start(opts)
 
     if sequential_startup then
         for i, server in ipairs(self.servers) do
-            server:box_config({replication = replication_settings[i]})
-            server.box_cfg.replication = replication_settings[i]
+            if replication_settings[i] ~= nil then
+                server:box_config({replication = replication_settings[i]})
+                server.box_cfg.replication = replication_settings[i]
+            end
         end
     end
 
```

Reply:
Ah, yes, I didn't understand the problem. I propose following patch, it will work both for starting cluster with already started servers (will not touch them) and for starting server where box_cfg.replication == nil (will still set it back to nil after starting)

```diff
diff --git a/test/luatest_helpers/cluster.lua b/test/luatest_helpers/cluster.lua
index 0d0390691..0be7e0ce4 100644
--- a/test/luatest_helpers/cluster.lua
+++ b/test/luatest_helpers/cluster.lua
@@ -85,6 +85,7 @@ function Cluster:start(opts)
     end

     local replication_settings = {}
+    local started_server = {}

     for i, server in ipairs(self.servers) do
         if not server.process then
@@ -93,13 +94,16 @@ function Cluster:start(opts)
                 server.box_cfg.replication = self.servers[1].net_box_uri
             end
             server:start({wait_for_readiness = sequential_startup})
+            started_server[i] = true
         end
     end

     if sequential_startup then
         for i, server in ipairs(self.servers) do
-            server:box_config({replication = replication_settings[i]})
-            server.box_cfg.replication = replication_settings[i]
+            if started_server[i] == true then
+                server:box_config({replication = replication_settings[i]})
+                server.box_cfg.replication = replication_settings[i]
+            end
         end
     end
```

Reply:
Looks good, thanks! I missed the other part of the problem.

Reply:
Thanks for review! I've squashed this patch and forced pushed PR branch.